### PR TITLE
Pia 3133 implement image cropping

### PR DESCRIPTION
**Pull request by vizaknaidavid**
This code should crop and show only the part of the image which is inside of the white brackets when taking a photo.

### CODE REVIEW

**Review by zladzeyka — COMMENTED** [review]
Thank you, David!
I've tested it with iPhone 13 Pro Max - it crops more than needed 🥲(no bottom navigation/with bottom navigation the same)

https://user-images.githubusercontent.com/3911816/197537449-a1374948-10cf-4cfc-bec6-63e135b832db.MP4

**Review by chriskryn — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/1632312/197544399-4161c6ec-7a27-43ae-8568-549b62671e5b.MP4

Unfortunately, the cropping doesn't seem to work properly on iPhone11 it takes more than it should. 
[Sugestion] @vizaknaidavid Could you add a comment somewhere in code that describes the algorithm. Then, It might be easier to spot some issue, suggest some changes ?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Extensions/CGRect.swift [hunk 1]
```diff
@@ -12,4 +12,11 @@ extension CGRect {
     var center: CGPoint {
         return CGPoint(x: midX, y: midY)
     }
+
+    func scaled(for scaleFactor: CGFloat) -> CGRect {
+        return CGRect(x: self.minX * scaleFactor,
```
[the remark this answers is not in the document]
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Extensions/CGRect.swift:17]: As you can see in the code, the origin is every time different from zero, so it needs to be scaled together with the dimensions

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift [hunk 2]
```diff
@@ -86,14 +86,25 @@ final public class GiniImageDocument: NSObject, GiniCaptureDocument {
...
         var updatedRect: CGRect
 
-        print("X: image.size.width \(image.size.width) image.size.height \(image.size.height)")
-
-
         if image.size.width < image.size.height {
-//            updatedRect = CGRect(x: 500, y: 20, width: 2000, height: 2800)
-            updatedRect = CGRect(x: 20, y: 500, width: 2800, height: 2000)
+            if UIDevice.current.isIpad {
+                updatedRect = CGRect(x: 350, y: 350, width: 2400, height: 3300)
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift]: [Clarification] @vizaknaidavid could you please add the details of why you use this specific Rect?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift [hunk 3]
```diff
@@ -86,14 +86,25 @@ final public class GiniImageDocument: NSObject, GiniCaptureDocument {
...
+            } else {
+                updatedRect = CGRect(x: 20, y: 500, width: 2800, height: 2000)
+            }
         } else {
-            updatedRect = CGRect(x: 500, y: 20, width: 2000, height: 2800)
+            if UIDevice.current.isIpad {
+                if image.imageOrientation == .up {
+                    updatedRect = CGRect(x: 700, y: 50, width: 2100, height: 2900)
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift]: [Suggestion] Could we extract these Rects in a separate func -> like calculateUpdatedRect() and do the calculation there with some inline comments 😊?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/3911816/197581259-e8b5356b-3e86-4d0d-be3d-d3b7b14767be.MP4

This is how it looks on iPhone SE - let's check tomorrow at the algorithm together 😉

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift (hunk 2 above)
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift]: This is an older version of the code, there are no more exact numbers there

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift (hunk 3 above)
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Models/GiniImageDocument.swift]: Already added the comments to this functiono, but we can also separate it in a different function

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/Camera2/Camera2ViewController.swift [hunk 4]
```diff
@@ -151,15 +149,15 @@ public final class Camera2ViewController: UIViewController, CameraScreen {
 
     private func layoutBottomNavigationBar(_ navigationBar: UIView) {
         if UIDevice.current.isIpad {
-            view.removeConstraints([iPadCameraFrameBottomConstraint])
+            view.removeConstraints([cameraPreviewBottomContraint])
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/Camera2/Camera2ViewController.swift:152]: [Clarification]: iPadCameraFrameBottomConstraint was used there, shall we use now cameraPreviewBottomContraint for iPhone and iPad?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift [hunk 5]
```diff
@@ -38,6 +38,14 @@ final class CameraPreviewViewController: UIViewController {
         spinner.hidesWhenStopped = true
         return spinner
     }()
+
+    lazy var cameraFrameView: UIView = {
+        let imageView = UIView()
+        imageView.layer.borderColor = UIColor.white.cgColor
```
Comment (zladzeyka) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift:44]: [Suggestion] Please use the color here for both modes: light and dark

**Review by zladzeyka — COMMENTED** [review]
Great job @vizaknaidavid! 👏🏻🎉
Could you please add some inline comments in the code? 😉

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift (hunk 5 above)
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift:44]: You mean to have the same color for both modes, or to have white/black regarding light/dark mode?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Extensions/UIImage.swift [hunk 6]
```diff
@@ -81,5 +81,71 @@ extension UIImage {
...
+            transform = transform.translatedBy(x: 0, y: size.height)
+            transform = transform.rotated(by: CGFloat(-Double.pi / 2))
+
+        case .up, .upMirrored:
+            break
+        @unknown default:
+            break
+        }
+
+        switch imageOrientation {
```
[the remark this answers is not in the document]
Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Extensions/UIImage.swift:114]: The camera of the phone writes the EXIF with orientation of the phone, but even on Apple's camera app, the images are saved with `.up` orientation. 
This extension is a solution from another project, which overrides the orientation and saves the image with `.up` orientation regardless of how it was shot. This won't affect image quality, resolution or anything. I will try to describe what the lines do.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift (hunk 5 above)
Comment (zladzeyka) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift:44]: We've agreed to add a frame color to the color palette.

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thank you for the comments. Looks good!
I've tested the cropping on the iPhone and IPad and noticed the issue in the landscape orientation, you can reproduce it on the simulator as well, just start the app in the landscape.
Simulator Screen Shot - iPad (9th generation) - 2022-10-28 at 15 01 45

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift [hunk 7]
```diff
@@ -73,6 +80,37 @@ final class CameraPreviewViewController: UIViewController {
         previewView.addGestureRecognizer(tapGesture)
         return previewView
     }()
+
+    private let iPadCameraPaneWidth: CGFloat = 124
+    private lazy var iPadLandscapeConstraints: [NSLayoutConstraint] = [
+        cameraFrameView.topAnchor.constraint(equalTo: view.topAnchor, constant: 16),
+        cameraFrameView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
```
Comment (chriskryn) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift:87]: [Suggestion] @vizaknaidavid Could you add some constant for the  numbers like '16', '1.414'?

**Review by chriskryn — COMMENTED** [review]
Looks good! I have just very small suggestion.

**Review by chriskryn — APPROVED** [review]
Looks good! :+1:

**Review by zladzeyka — APPROVED** [review]
Great success 👏🏻😊👍🏻

**Pull request merged**